libraries/ArduinoOTA/ArduinoOTA.cpp

```diff
@@ -320,18 +314,27 @@ void ArduinoOTAClass::_runUpdate() {
     client.flush();
     delay(1000);
     client.stop();
-#ifdef OTA_DEBUG
-    OTA_DEBUG.printf("Update Success\n");
-#endif
+    OTA_DEBUG_PRINTF("Update Success\n");
     if (_end_callback) {
       _end_callback();
     }
     if(_rebootOnSuccess){
-#ifdef OTA_DEBUG
-    OTA_DEBUG.printf("Rebooting...\n");
-#endif
+      OTA_DEBUG_PRINTF("Rebooting...\n");
       //let serial/network finish tasks that might be given in _end_callback
       delay(100);
+      if (_eraseConfig) {
+        eraseConfigAndReset();  // returns on failure
+        //C What is the best action to take on failure?
+        //C  1) On failure, we could invalidate eboot_command buffer -
+        //C     aborting the flash update.
+        //C  2) Just ignore it and restart.
+        //C  3) Retry forever
+        if (_error_callback) {
+          _error_callback(OTA_ERASE_SETTINGS_ERROR);
+        }
```

Comment:
At failure, there are no good choices. 
I picked 3 to retry forever. 
If they power cycle, the download is aborted.

Reply:
Why a separate condition & callback though? We can retrieve error id

Reply:
I don't understand your reference. I am using the existing error callback with a new error number to identify the situation.

Reply:
(review UI sent a draft :/)

Why even mention it? Since we want this to be done, probably unroll is a better option than leave things in an indeterminate state?
Another part - it is the only state that continues execution. We still have eboot part like you said, even after ::end()ing the instance (suppose that from user pov it is an option to recover).

Reply:
https://github.com/esp8266/Arduino/blob/0e1059d60f3f3799589b92469609cf485e2b275c/libraries/ArduinoOTA/ArduinoOTA.cpp#L328-L346

When enabling erase WiFi settings, the user does so by selecting the fail option ignore or abort. Does that satisfy your concerns?